cg/meta/orders/api.py

```diff
@@ -36,25 +38,27 @@ def __init__(
         self.ticket_handler = ticket_handler
         self.submitter_registry = submitter_registry
 
-    def submit(self, project: OrderType, order_in: OrderIn, user_name: str, user_mail: str) -> dict:
+    def submit(
+        self, order_type: OrderType, raw_order: dict, user_name: str, user_mail: str
+    ) -> dict:
         """Submit a batch of samples.
 
         Main entry point for the class towards interfaces that implements it.
         """
-        submit_handler = self.submitter_registry.get_order_submitter(project)
-        submit_handler.order_validation_service.validate_order(order_in)
+        submit_handler: OrderSubmitter = self.submitter_registry.get_order_submitter(order_type)
+        order: Order = submit_handler.order_validation_service.parse_and_validate(raw_order)
         # detect manual ticket assignment
-        ticket_number: str | None = self.ticket_handler.parse_ticket_number(order_in.name)
+        ticket_number: str | None = self.ticket_handler.parse_ticket_number(order.name)
         if not ticket_number:
```

Comment:
So for the future we might want to remove this check and deprecate the connect to order support.

Reply:
Indeed. But I don't necessarily want to include that issue as part of this project